File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        s = s + '1'
        cnt1 = 0
        ans = 0

        n_minus_1 = len(s) - 1
        s_local = s

        for i in range(n_minus_1):
            c = s_local[i]
            if c == '1':
                cnt1 += 1
            else:
                if s_local[i+1] == '1':
                    ans += cnt1

        return ans

File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.cpp
class Solution {
public:
    int maxOperations(string s) {
        int cntOnes = 0;
        int ans = 0;
        s.push_back('1');
        for (int i = 0; i < s.size() - 1; i++) {
            if (s[i] == '1') {
                cntOnes++;
            }
            if (s[i] == '0' &&  s[i+1] == '1') {
                ans += cntOnes;
            } 
        }
        return ans;
    }
};